src/runtime/slot-consumer.ts

```diff
@@ -15,6 +15,23 @@ import {Description} from './description.js';
 import {SlotConnection} from './recipe/slot-connection.js';
 import {HostedSlotContext, ProvidedSlotContext, SlotContext} from './slot-context.js';
 
+export interface Model {
+  id: string;
+}
+
+export interface Content {
+  templateName?: string | Map<string, string>;
+  model?: {models: Model, hash: string};
+  descriptions?: Map<string, Description>;
+  template?: string | Map<string, {}>;
+}
+
+export interface Render {
+  container?: {};
+  model?: {items: {models: Model[]}, subId: {}, message: {}};
```

Comment:
i don't think you can declare this about `model` - it doesn't have to contain any of these, it is just a JSON. 
It so happens that the set slot ones contain `items` and `subId`, but this is more of a silent assumption and a design flaw rather than a requirements (especially the `items`).

Reply:
Thanks Maria, had another shot. Won't be able to have another till at least Tuesday so no rush.